python/tests/backends/test_IonQ.py

```diff
@@ -35,7 +36,10 @@ def startUpMockServer():
     # Launch the Mock Server
     p = Process(target=startServer, args=(port,))
     p.start()
-    time.sleep(1)
+
+    if not check_server_connection(port):
+        p.terminate()
+        pytest.exit("Mock server did not start in time, skipping tests.")
```

Comment:
Untested, but I think this is what is needed in order to show it as a test failure:
```suggestion
        pytest.exit("Mock server did not start in time, skipping tests.", returncode=1)
```